indico/htdocs/js/indico/modules/events/cloning.js

```diff
@@ -0,0 +1,129 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /* global getFormParams:false */
+
+
+(function(global) {
+    'use strict';
+
+    global.setupCloneDialog = function setupCloneDialog() {
+        var $formContainer = $('#event-clone-form-container');
+        var $form = $('#event-clone-form');
+        var $eventCount = $('#event-count').hide();
+        var $eventList = $form.find('.event-list');
+        var $cloneErrors = $('#clone-errors').hide();
+        var step = $formContainer.data('step');
+
+        var clonerDependencies = $formContainer.data('clonerDependencies');
+
+
+        function errorToHTML(error) {
+            if (typeof error == 'string') {
+                return error;
+            } else {
+                return $('<div>').append(error.map(function(item) {
+                    var label = $('<strong>').append(item[0]);
+                    var items = $('<ul>').append(item[1].map(function(message) {
+                        return $('<li>').text(message);
+                    }));
+                    return $('<div>').append(label, items);
+                }));
+            }
+        }
+
+        function updateCount() {
+            var $cloneButton = $('.clone-action-button');
+
+            $.ajax({
+                url: $formContainer.data('previewUrl'),
+                method: 'POST',
+                data: getFormParams($form),
+                success: function(data) {
+                    if (data.success) {
+                        var $countNumber = $eventCount.find('.count');
+                        $countNumber.text(data.count);
+                        $cloneErrors.hide();
+                        $eventCount.show();
+
+                        if (data.count) {
+                            $eventList.show();
+                            $cloneButton.prop('disabled', false);
+                        } else {
+                            $eventList.hide();
+                            $cloneButton.prop('disabled', true);
+                        }
+                        $eventCount.data('eventDates', data.dates);
+                    } else {
+                        $cloneErrors.show().find('.message-text').html(errorToHTML(data.error.message));
+                        $eventCount.hide();
+                    }
+                }
+            });
+        }
+
+        if (step === 4) {
+            $form.on('change', 'select,input', updateCount);
+            $form.on('keyup', 'input', function(e) {
+                e.preventDefault();
+                updateCount();
+            });
+        }
+
+        if ($eventCount.length) {
+            setTimeout(updateCount, 100);
```

Comment:
Basically, when the form loads some of the fields are set from JavaScript. If we immediately send the request, we will send wrong information to the server. I know it's hacky, but I can't see a better solution right now. Suggestions are welcome!